pkg/machine/hyperv/vsock/vsock.go

```diff
@@ -202,10 +206,115 @@ func NewHVSockRegistryEntry(machineName string, purpose HVSockPurpose) (*HVSockR
 		Port:        port,
 		MachineName: machineName,
 	}
+
+	return &r, nil
+}
+
+// NewHVSockRegistryEntry is a constructor to make a new registry entry in Windows.  After making the new
+// object, it calls the add() method to *actually* add it to the Windows registry.
+func NewHVSockRegistryEntry(machineName string, purpose HVSockPurpose) (*HVSockRegistryEntry, error) {
+	r, err := CreateHVSockRegistryEntry(machineName, purpose)
+	if err != nil {
+		return nil, err
+	}
+
 	if err := r.Add(); err != nil {
 		return nil, err
 	}
-	return &r, nil
+
+	return r, nil
+}
+
+// AddHVSockRegistryEntries allows to *actually* add multiple registry entries to the Windows registry
+//
+// When init a podman machine with hyperV, we have to add registry entries under
+// HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Virtualization\GuestCommunicationServices
+// so that host/guest can communicate using the vsock protocol.
+// The location in the registry is fixed and cannot be changed, this is why elevated privileges are required.
+// Entries under HKEY_LOCAL_MACHINE can just be created by an admin.
+//
+// To ensure compatibility between Linux guests and the Windows host, a "well-known GUID" is needed.
+// The format, as described by HV_GUID_VSOCK_TEMPLATE, must be <port>-facb-11e6-bd58-64006a7986d3
+// where <port> are the first four bytes ("00000000")  reserved for the port number
+// (e.g  "00000ac9" represents port 2761 -> hexadecimal AC9 is decimal 2761)
+//
+// This func supports bulk insertion so the user is asked for elevated rights only once
+func AddHVSockRegistryEntries(entries []HVSockRegistryEntry) error {
```

Comment:
Are `(hv *HVSockRegistryEntry) Add()`, `NewHVSockRegistryEntry()` etc...still used? Because otherwise we should remove them.